### ISSUE TITLE
Output directory parameter for testing

### ISSUE THREAD

**Issue by egeerardyn** [matlab2tikz/matlab2tikz#816]
Currently (e.g. as discussed in #321, #326 and #814), there are some limitations to the testing framework that impede quick and easy iterations for us developers.

We should be able to set the output directory for use in testing. That would allow
   * parallel testing on a single checkout using different environments (mainly of importance on developer machines) without nasty race conditions
   * to move complexity from developer or reporting tools towards the testing suite. Currently, this complexity is basically file management (moving those files around by hand, getting outdated files out of the way, ...), but this is cumbersome, error-prone and mind-numbing.

**Comment by miscco** [comment]
Could we add that as an ipp parameter and do some shenanigans in execute_tikz_stage like:

```
function [status] = execute_tikz_stage(status, ipp)
% test stage: TikZ file generation
    testNumber = status.index;

    dataDir = ipp.dataDir;
    % Check if dataDir does exist
    if(exist(['data/',dataDir], 'dir')==0)
        system(['mkdir data/', dataDir]);
    end
    gen_tex = sprintf(['data/',dataDir,'/test%d-converted.tex'], testNumber);
    gen_pdf  = sprintf(['data/',dataDir,'/converted/test%d-converted.pdf'], testNumber);
    cleanfigure_time = NaN;
    m2t_time = NaN;

    % now, test matlab2tikz
    try
        cleanfigure_time = tic;
        cleanfigure(status.extraCleanfigureOptions{:});
        cleanfigure_time = toc(cleanfigure_time);

        m2t_time = tic;
        matlab2tikz('filename', gen_tex, ...
            'showInfo', false, ...
            'checkForUpdates', false, ...
            'dataPath', ['data/',dataDir,'], ...
            'standalone', true, ...
            ipp.Results.extraOptions{:}, ...
            status.extraOptions{:} ...
            );
        m2t_time = toc(m2t_time);
    catch %#ok
        e = lasterror('reset'); %#ok
        % Remove (corrupted) output file. This is necessary to avoid that the
        % Makefile tries to compile it and fails.
        delete(gen_tex)
        [status.tikzStage] = errorHandler(e);
    end
    status.tikzStage.texFile = gen_tex;
    status.tikzStage.pdfFile = gen_pdf;
    status.tikzStage.m2t_time = m2t_time;
    status.tikzStage.cleanfigure_time = cleanfigure_time;
end

**Comment by egeerardyn** [comment]
@miscco I don't fully understand what you are trying to say.

I think you meant something in connection to #818 where this feature is implemented. The output directory is given there as a parameter; the data directory is then specified implicitly by means of the template directory and the code.

**Comment by miscco** [comment]
Yepp, wrong tab

**Comment by egeerardyn** [comment]
This has been implemented in #818 where I forgot to add a "fixes ..." phrase.
